### PR TITLE
Improve documentation around version strategies

### PR DESCRIPTION
**Pull request by selfdocumentingcode**
## Description
- Improve listing of version strategies on "How it works" docs page
  - Update names to reflect recent code changes
  - Replace (incorrect) examples with a link to the main version strategies list
  - Concisely describe `Mainline` and `TrackReleaseBranches` version strategies
- Improve listing of version strategies on "Configuration" docs page
  - Replace `TrunkBased` strategy listing with `Mainline` 
- Fix Edit Diagram link on "How it works" docs page
- Update `docs\readme.me` build instructions to use correct target name for docs preview

## Related Issue
#4219 

## Motivation and Context
I'm trying to understand how GitVersion works and the documentation makes it difficult at times.

## How Has This Been Tested?
I built and ran a preview of /docs locally. 

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by arturcic** [comment]
> * Remove unused file `docs\input\docs\references\mdsource\configuration.source.md`

This one is actually used, and we have tests that check the content of the `configuration.md` based on the `configuration.source.md`. See https://github.com/GitTools/GitVersion/pull/4104

File: build/docs/BuildLifetime.cs [hunk 1]
```diff
@@ -25,8 +25,7 @@ public override void Setup(BuildContext context, ISetupContext info)
                 { "BaseEditUrl", "https://github.com/gittools/GitVersion/tree/main/docs/input/" },
                 { "SourceFiles", context.MakeAbsolute(Paths.Src) + "/**/{!bin,!obj,!packages,!*.Tests,!GitTools.*,}/**/*.cs" },
                 { "Title", "GitVersion" },
-                { "IncludeGlobalNamespace", false },
-                { "IgnoreFolders", "**/mdsource" }
```
Comment (arturcic) [inline build/docs/BuildLifetime.cs:29]: This should be kept

**Review by arturcic — CHANGES_REQUESTED** [review]

**Comment by selfdocumentingcode** [comment]
I see. That feature is not mentioned anywhere in the readme file nor did any tests failed after removing it. 

I will, of course, remove the change from the PR. 
Thanks for pointing it out!

**Comment by arturcic** [comment]
> I see. That feature is not mentioned anywhere in the readme file nor did any tests failed after removing it. 
> 
> I will, of course, remove the change from the PR. 
> Thanks for pointing it out! 
> 
> Edit: @arturcic I am right to assume that for this particular change I'm making to the `configuration.md` file, i.e. unrelated to the GitVersion workflows, I have to also make the same change in the `configuration.source.md` file in order for the change in `configuration.md` to not get reverted the next time the `mkdocs.yml` workflow runs? 

Yes, please update the source file instead, as the `configuration.md` is updated from the source

File: docs/readme.md
Comment (arturcic) [inline docs/readme.md:1]: can you also include the details for MarkdownSnippets that converts the `*.source.md` into `*.md` files in this readme?
This tool is used to embed some files into md files

https://github.com/GitTools/GitVersion/blob/main/.github/workflows/mkdocs.yml#L24-L28

Comment (selfdocumentingcode) [reply docs/readme.md:1]: It's your feature. I'm sure you can provide a better documentation than I ever could.

**Comment by arturcic** [comment]
@HHobeck can you check the changes for the documentations?

**Comment by HHobeck** [comment]
> @HHobeck can you check the changes for the documentations?

Yes, I try to do the review tomorrow.

File: docs/input/docs/learn/how-it-works.md [hunk 2]
```diff
@@ -19,51 +19,54 @@ GitVersion has three distinct steps for calculating versions in v3.
...
+* `MergeMessage` - Finds version numbers from merge messages
     (e.g., `Merge 'release/3.0.0' into 'main'` will return `3.0.0`)
-* `FallbackBaseVersionStrategy` - Always returns 0.0.0 and will be used for
+* `Fallback` - Always returns 0.0.0 and will be used for
     calculating the next version which is dependent on the increment strategy of
  the effected branch (e.g. on main the next version is 0.0.1 or on develop it is 0.1.0)
+* `TrackReleaseBranches` - Considers the base version extracted from release branches when
+calculating the next version for branches configured with `track-release-branches: true`
+(part of default configuration for `develop` branch in `GitFlow` workflow)
+* `Mainline` - Increments the version on every commit for branches configured with `mode: Mainline`
```
Comment (HHobeck) [inline docs/input/docs/learn/how-it-works.md:55]: mode: Mainline is not valid anymore. I think is-main-branch: true would be better

Comment (selfdocumentingcode) [reply docs/input/docs/learn/how-it-works.md]: Understood! 
And fixed!

**Review by HHobeck — COMMENTED** [review]
May I ask you to document this? https://github.com/GitTools/GitVersion/discussions/4248

**Comment by selfdocumentingcode** [comment]
> May I ask you to document this? #4248

Sure! Check my latest commits.

**Comment by arturcic** [comment]
@HHobeck if there are no additional changes that need to be added, please merge this PR

**Review by HHobeck — APPROVED** [review]

**Comment by HHobeck** [comment]
> @HHobeck if there are no additional changes that need to be added, please merge this PR

Not sure why just the auto merge option is shown

**Comment by arturcic** [comment]
I disabled auto-merge, but because there are some required actions (that are not run because only docs where changed) it's in pending state, so I will just merge

**Pull request merged**